cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -197,6 +199,13 @@ func getAPIServerCommand(cfg *kubeadmapi.ClusterConfiguration, localAPIEndpoint
 		defaultArguments["feature-gates"] = fmt.Sprintf("%s=%t", features.IPv6DualStack, enabled)
 	}
 
+	// TODO: remove after BoundServiceAccountTokenVolume is GA.
+	if enabled, ok := cfg.FeatureGates["AllBeta"]; ok && !enabled || strings.Contains(cfg.APIServer.ExtraArgs["feature-gates"], "AllBeta=false") {
```

Comment:
kubeadm does not use the same list of feature gates as other k8s components. it only has these:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/features/features.go#L30-L35

this branch should be removed, and users that wish to opt-out of BoundServiceAccountTokenVolume should manually delete the flags `service-account-issuer`, `service-account-signing-key-file` from the kube-apiserver manifests.

Reply:
I agree this branch is not great ... this is probably added here because of the ga-only conformance job which sets AllBeta=false and expects clusters to start and pass conformance tests.

The following features are beta:
1. TokenRequest (enables the API server to serve the /tokens endpoint and accept the signing-key/issuer flags)
2. TokenRequestProjection (enables pods to request projected token volumes)
3. BoundServiceAccountTokenVolume (enables the API server to inject service account tokens as projected token volumes by default)

Right now, the ga-only conformance job disables all three features with AllBeta=false. That means the API server complains about the signing-key/issuer flags. @mikedanese Moving TokenRequest/TokenRequestProjection to GA now seems reasonable to me, and would allow deployers like kubeadm to start populating the signing key and issuer flags.

Reply:
thanks for the details!

> Moving TokenRequest/TokenRequestProjection to GA now seems reasonable to me, and would allow deployers like kubeadm to start populating the signing key and issuer flags.

ok, if not the branch can be adapted to something like the following, to conform the ga-only jobs without patches:

```
// TODO: remove after BoundServiceAccountTokenVolume is GA.
// ------> please add link to tracking issue here <------
if strings.Contains(cfg.APIServer.ExtraArgs["feature-gates"], "AllBeta=false") {
	delete(defaultArguments, "service-account-issuer")
	delete(defaultArguments, "service-account-signing-key-file")
}
```

but i cannot seem to be able to find the tracking issue for:
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/auth/bound-service-account-tokens.md

Reply:
> @mikedanese Moving TokenRequest/TokenRequestProjection to GA now seems reasonable to me, and would allow deployers like kubeadm to start populating the signing key and issuer flags

This sounds good to me.